taskvine/src/manager/vine_manager.c

```diff
@@ -2698,7 +2703,13 @@ struct rmsummary *vine_manager_choose_resources_for_task(struct vine_manager *q,
 			/* worker's disk is shared evenly among tasks that are not running,
 			 * thus the proportion is modified by the current overcommit
 			 * multiplier */
-			limits->disk = MAX(1, MAX(limits->disk, floor(available_disk * max_proportion / q->resource_submit_multiplier)));
+			limits->disk = MAX(1, MAX(limits->disk, floor(disk_total * max_proportion / q->resource_submit_multiplier)));
+
+			/* when the proportional resource allocation is larger than the available, use the available instead */
+			limits->cores = MIN(limits->cores, cores_available);
```

Comment:
As discussed, we may only want to update the proportion when it is larger than the *_available, then only update it to  allocated/2.  If that would not fit in *_available, that's ok.  We don't want a tasks to end up with 1MB of memory/disk.